### PR TITLE
Added 0.60.0 release notes

### PR DESCRIPTION
**Pull request by kc611**
As titled, this PR adds changelog for `0.60.0` release.

### CODE REVIEW

File: docs/source/release/0.60.0-notes.rst [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+(`PR-#9482 <https://github.com/numba/numba/pull/9482>`__)
+
+
+NumPy Support
+~~~~~~~~~~~~~
+
+Added support for ``np.size()``
+-------------------------------
+
+Added ``np.size()`` support for numpy, which was previously unsupported.
```
Comment (esc) [inline docs/source/release/0.60.0-notes.rst:127]:
```suggestion
Added ``np.size()`` support for NumPy, which was previously unsupported.
```

File: docs/source/release/0.60.0-notes.rst [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+---------------------------------------------
+
+Support ``math.nextafter`` in nopython mode.
+
+(`PR-#9438 <https://github.com/numba/numba/pull/9438>`__)
+
+Add support for parfor binop reductions.
+----------------------------------------
+
+Previously, only operations with inplace operations like += could be used as reductions in pranges.  Now, with this PR, binop reductions of the form a = a binop b can be used.
```
Comment (esc) [inline docs/source/release/0.60.0-notes.rst:89]:
```suggestion
Previously, only operations with inplace operations like `+=` could be used as reductions
in `prange`s.  Now, with this PR, binop reductions of the form `a = a binop b` can be used.
```

**Review by esc — CHANGES_REQUESTED** [review]
Some markup and spelling nitpicks.

But importantly, the PR is missing the list of PRs and Authors.

**Comment by esc** [comment]
> But importantly, the PR is missing the list of PRs and Authors.

You'll need the script at: https://github.com/numba/numba/blob/main/maint/gitlog2changelog.py

**Comment by kc611** [comment]
I used the command 
```
python maint/gitlog2changelog.py --repo=numba/numba --beginning="0.59.1" --token=<Personal Access Token> --digits=4 --summary
```

to generate the following list:

```
Pull-Requests:

* PR `#8984 <https://github.com/numba/numba/pull/8984>`_: Support @gufunc inside @jit (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
* PR `#9239 <https://github.com/numba/numba/pull/9239>`_: ufunc.at (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
* PR `#9274 <https://github.com/numba/numba/pull/9274>`_: CUDA: Add support for compilation to LTO-IR (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9364 <https://github.com/numba/numba/pull/9364>`_: Release notes fixes for appropriate Towncrier header underlines (`kc611 <https://github.com/kc611>`_)
* PR `#9367 <https://github.com/numba/numba/pull/9367>`_: Document release notes generation (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9369 <https://github.com/numba/numba/pull/9369>`_: Fix release notes link in bug report template (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9378 <https://github.com/numba/numba/pull/9378>`_: Add `float(<string literal>)` ctor (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
* PR `#9394 <https://github.com/numba/numba/pull/9394>`_: fix `TypeError: '_DeviceContextManager' object is not callable` (`i7878 <https://github.com/i7878>`_)
* PR `#9402 <https://github.com/numba/numba/pull/9402>`_: Doc updates for 0.59 final (`sklam <https://github.com/sklam>`_)
* PR `#9403 <https://github.com/numba/numba/pull/9403>`_: Fix test isolation for stateful configurations in the testsuite (`sklam <https://github.com/sklam>`_)
* PR `#9404 <https://github.com/numba/numba/pull/9404>`_: Fix skipped test stderr change for Python 3.12.1. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9407 <https://github.com/numba/numba/pull/9407>`_: Fix incorrect cycle detection (`sklam <https://github.com/sklam>`_)
* PR `#9411 <https://github.com/numba/numba/pull/9411>`_: Doc updates for 0.59.0 final. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9416 <https://github.com/numba/numba/pull/9416>`_: Add math.log2 support (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
* PR `#9417 <https://github.com/numba/numba/pull/9417>`_: Add np.log* bindings for CUDA (`gmarkall <https://github.com/gmarkall>`_ `guilhermeleobas <https://github.com/guilhermeleobas>`_)
* PR `#9425 <https://github.com/numba/numba/pull/9425>`_: Post release for 0.59.0 (`sklam <https://github.com/sklam>`_)
* PR `#9436 <https://github.com/numba/numba/pull/9436>`_: Add timing and junit xml output to testsuite (`sklam <https://github.com/sklam>`_)
* PR `#9437 <https://github.com/numba/numba/pull/9437>`_: Remove dependencies between Numba's Cpython and NumPy module (`kc611 <https://github.com/kc611>`_)
* PR `#9438 <https://github.com/numba/numba/pull/9438>`_: Add math.nextafter support for nopython mode. (`groutr <https://github.com/groutr>`_)
* PR `#9445 <https://github.com/numba/numba/pull/9445>`_: Fix #9427 np.searchsorted on datetime64 (`sklam <https://github.com/sklam>`_)
* PR `#9447 <https://github.com/numba/numba/pull/9447>`_: Fix Issue #9432, caching of kernels using target-specific overloads (such as CG sync) (`sklam <https://github.com/sklam>`_ `gmarkall <https://github.com/gmarkall>`_)
* PR `#9448 <https://github.com/numba/numba/pull/9448>`_: Fix `np.searchsorted` regression (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
* PR `#9449 <https://github.com/numba/numba/pull/9449>`_: Remove deprecated CondaEnvironment@1 (`sklam <https://github.com/sklam>`_)
* PR `#9450 <https://github.com/numba/numba/pull/9450>`_: Fix gpuci versions (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9454 <https://github.com/numba/numba/pull/9454>`_: Don't attempt to register overloads that aren't for this target in `BaseContext` and related fixes (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9455 <https://github.com/numba/numba/pull/9455>`_: Support datetime types in `isinstance()` (`sklam <https://github.com/sklam>`_)
* PR `#9456 <https://github.com/numba/numba/pull/9456>`_: Update release checklist (`sklam <https://github.com/sklam>`_)
* PR `#9459 <https://github.com/numba/numba/pull/9459>`_: Support pep695 type param syntax (`sklam <https://github.com/sklam>`_)
* PR `#9466 <https://github.com/numba/numba/pull/9466>`_: Numpy 2.0 binary support testing (`kc611 <https://github.com/kc611>`_)
* PR `#9468 <https://github.com/numba/numba/pull/9468>`_: adding git-copy.py script (`esc <https://github.com/esc>`_)
* PR `#9482 <https://github.com/numba/numba/pull/9482>`_: Add support for `sys.monitoring` events. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9487 <https://github.com/numba/numba/pull/9487>`_: Add CUDA 12 conda installation docs (`bdice <https://github.com/bdice>`_ `gmarkall <https://github.com/gmarkall>`_)
* PR `#9488 <https://github.com/numba/numba/pull/9488>`_: Update overview.rst (`jftsang <https://github.com/jftsang>`_)
* PR `#9491 <https://github.com/numba/numba/pull/9491>`_: Fix non-deterministic bug caused by unstableness in SSA  (`sklam <https://github.com/sklam>`_ `loicdtx <https://github.com/loicdtx>`_)
* PR `#9502 <https://github.com/numba/numba/pull/9502>`_: Post release task for 0.59.1 (`sklam <https://github.com/sklam>`_)
* PR `#9504 <https://github.com/numba/numba/pull/9504>`_: added np.size() overload and added tests (`shourya5 <https://github.com/shourya5>`_)
* PR `#9521 <https://github.com/numba/numba/pull/9521>`_: Support binop reduction. (`DrTodd13 <https://github.com/DrTodd13>`_)
* PR `#9531 <https://github.com/numba/numba/pull/9531>`_: Module pass manager: Don't add passes for unsupported LLVM versions (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9532 <https://github.com/numba/numba/pull/9532>`_: Fix hoisting bug to exclude variables used in containers or calls. (`DrTodd13 <https://github.com/DrTodd13>`_)
* PR `#9539 <https://github.com/numba/numba/pull/9539>`_: Revert PR #6870 `numba.core.retarget` (`sklam <https://github.com/sklam>`_)
* PR `#9549 <https://github.com/numba/numba/pull/9549>`_: Make new_style the default error capturing mode (`sklam <https://github.com/sklam>`_ `gmarkall <https://github.com/gmarkall>`_)

Total PRs: 41

Authors:

* `bdice <https://github.com/bdice>`_
* `DrTodd13 <https://github.com/DrTodd13>`_
* `esc <https://github.com/esc>`_
* `gmarkall <https://github.com/gmarkall>`_
* `groutr <https://github.com/groutr>`_
* `guilhermeleobas <https://github.com/guilhermeleobas>`_
* `i7878 <https://github.com/i7878>`_
* `jftsang <https://github.com/jftsang>`_
* `kc611 <https://github.com/kc611>`_
* `loicdtx <https://github.com/loicdtx>`_
* `shourya5 <https://github.com/shourya5>`_
* `sklam <https://github.com/sklam>`_
* `stuartarchibald <https://github.com/stuartarchibald>`_

Total authors: 13
```
 But had to prune the list since some of the PRs listed were also included in the `0.59.1` and `0.59.0` changelog.

File: docs/source/release/0.60.0-notes.rst [hunk 3]
```diff
@@ -0,0 +1,206 @@
+
+Version 0.60.0 (Release Date)
```
[the remark this answers is not in the document]
Comment (kc611) [reply docs/source/release/0.60.0-notes.rst]: I have added tommorow's date for now. We can edit it later as required.

**Comment by esc** [comment]
> But had to prune the list since some of the PRs listed were also included in the `0.59.1` and `0.59.0` changelog.

This is to be expected. The script works by looking at given merge commits and using a regular expression on the commit message to work out which PR it was. The `--beginning` argument of the script is the starting point, usually a `dev` tag. The problem is that merges will be made to `main` after the `dev` tag has been set. For example during a patch release, PR are merged to `main` and then backported to the `release` branch. This means that they will show up with this script and the resulting list will always need to be pruned manually.

**Comment by esc** [comment]
> I used the command
> 
> ```
> python maint/gitlog2changelog.py --repo=numba/numba --beginning="0.59.1" --token=<Personal Access Token> --digits=4 --summary

Using the tag for the patch release is probably not the best idea here, since that is usually made on a release branch. Using the `dev` tags will be preferred.

**Comment by kc611** [comment]
Putting in the dev tag `0.59.0dev0` as the beginning give the following list of PRs:

```
Pull-Requests:

* PR `#8984 <https://github.com/numba/numba/pull/8984>`_: Support @gufunc inside @jit (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
* PR `#8990 <https://github.com/numba/numba/pull/8990>`_: Removed extra block copying in InlineWorker (`kc611 <https://github.com/kc611>`_)
* PR `#9048 <https://github.com/numba/numba/pull/9048>`_: Dynamically allocate parfor schedule. (`DrTodd13 <https://github.com/DrTodd13>`_)
* PR `#9058 <https://github.com/numba/numba/pull/9058>`_: Fix gufunc with multiple outputs (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
* PR `#9089 <https://github.com/numba/numba/pull/9089>`_: Fix segfault on passing `None` for args in PythonAPI.call (`hellozee <https://github.com/hellozee>`_)
* PR `#9101 <https://github.com/numba/numba/pull/9101>`_: Add misc script to find missing towncrier news files (`sklam <https://github.com/sklam>`_)
* PR `#9123 <https://github.com/numba/numba/pull/9123>`_: Implement most ufunc attributes and ufunc.reduce (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
* PR `#9126 <https://github.com/numba/numba/pull/9126>`_: Add support for np.indices() (`KrisMinchev <https://github.com/KrisMinchev>`_)
* PR `#9140 <https://github.com/numba/numba/pull/9140>`_: Add support for Polynomial class (`KrisMinchev <https://github.com/KrisMinchev>`_)
* PR `#9141 <https://github.com/numba/numba/pull/9141>`_: Add support for `as_series()` from `np.polynomial.polyutils` and `polydiv()`, `polyint()`, `polyval()` from `np.polynomial.polynomial` (`KrisMinchev <https://github.com/KrisMinchev>`_)
* PR `#9142 <https://github.com/numba/numba/pull/9142>`_: Removed out of date comment handled by PR#8338 (`njriasan <https://github.com/njriasan>`_)
* PR `#9144 <https://github.com/numba/numba/pull/9144>`_: Fix error when literal is wrongly propagated in a PHI node (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
* PR `#9148 <https://github.com/numba/numba/pull/9148>`_: bump llvmdev dependency to 0.42.0dev for next development cycle (`esc <https://github.com/esc>`_)
* PR `#9149 <https://github.com/numba/numba/pull/9149>`_: update release checklist post 0.58.0rc1 (`esc <https://github.com/esc>`_)
* PR `#9152 <https://github.com/numba/numba/pull/9152>`_: Fix old_style error capturing deprecation warnings (`sklam <https://github.com/sklam>`_)
* PR `#9154 <https://github.com/numba/numba/pull/9154>`_: Add support for np.unwrap() (`KrisMinchev <https://github.com/KrisMinchev>`_)
* PR `#9155 <https://github.com/numba/numba/pull/9155>`_: Remove unused test.cmd (`sklam <https://github.com/sklam>`_)
* PR `#9159 <https://github.com/numba/numba/pull/9159>`_: Fix uncaught exception in `find_file()` (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9168 <https://github.com/numba/numba/pull/9168>`_: fix the `get_template_info` method in `overload_method` template (`sklam <https://github.com/sklam>`_ `dlee992 <https://github.com/dlee992>`_)
* PR `#9169 <https://github.com/numba/numba/pull/9169>`_: Update `np.digitize` handling of np.nan bin edge(s) (`rjenc29 <https://github.com/rjenc29>`_)
* PR `#9170 <https://github.com/numba/numba/pull/9170>`_: Fix an inappropriate test expression to remove a logical short circuit (`munahaf <https://github.com/munahaf>`_)
* PR `#9171 <https://github.com/numba/numba/pull/9171>`_: Fix the implementation of a special method (`munahaf <https://github.com/munahaf>`_)
* PR `#9173 <https://github.com/numba/numba/pull/9173>`_: Towncrier fixups (Continue #9158 and retarget to main branch) (`sklam <https://github.com/sklam>`_ `esc <https://github.com/esc>`_)
* PR `#9181 <https://github.com/numba/numba/pull/9181>`_: Remove extra decrefs in RNG (`sklam <https://github.com/sklam>`_)
* PR `#9189 <https://github.com/numba/numba/pull/9189>`_: Align `searchsorted` behaviour with numpy (`rjenc29 <https://github.com/rjenc29>`_)
* PR `#9190 <https://github.com/numba/numba/pull/9190>`_: Fix issue with incompatible multiprocessing context in test. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9191 <https://github.com/numba/numba/pull/9191>`_: Add a Numba power-on-self-test script and use in CI. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9205 <https://github.com/numba/numba/pull/9205>`_: release notes and version support updates from release0.58 branch (`esc <https://github.com/esc>`_)
* PR `#9220 <https://github.com/numba/numba/pull/9220>`_: Support passing arbitrary flags to NVVM (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9223 <https://github.com/numba/numba/pull/9223>`_: CUDA: Add support for compiling device functions with C ABI (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9227 <https://github.com/numba/numba/pull/9227>`_: Support NumPy 1.26 (PR aimed at review / merge) (`gmarkall <https://github.com/gmarkall>`_ `Tialo <https://github.com/Tialo>`_)
* PR `#9228 <https://github.com/numba/numba/pull/9228>`_: Fix #9222 - Don't replace `.` with `_` in func arg names in inline closures (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9235 <https://github.com/numba/numba/pull/9235>`_: CUDA: Make `grid()` and `gridsize()` use 64-bit integers (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9236 <https://github.com/numba/numba/pull/9236>`_: Fixes numba/numba#9234 (`SridharCR <https://github.com/SridharCR>`_)
* PR `#9239 <https://github.com/numba/numba/pull/9239>`_: ufunc.at (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
* PR `#9244 <https://github.com/numba/numba/pull/9244>`_: Fix Recursion error in parfor lookup (`sklam <https://github.com/sklam>`_)
* PR `#9245 <https://github.com/numba/numba/pull/9245>`_: Fix slow InlineInlineable (`sklam <https://github.com/sklam>`_)
* PR `#9246 <https://github.com/numba/numba/pull/9246>`_: Support for Python 3.12 (`stuartarchibald <https://github.com/stuartarchibald>`_ `esc <https://github.com/esc>`_ `kc611 <https://github.com/kc611>`_)
* PR `#9249 <https://github.com/numba/numba/pull/9249>`_: add support for checking dtypes equal (`saulshanabrook <https://github.com/saulshanabrook>`_)
* PR `#9255 <https://github.com/numba/numba/pull/9255>`_: Fix SSA to consider variables whose use is not dominated by the definition (`sklam <https://github.com/sklam>`_)
* PR `#9258 <https://github.com/numba/numba/pull/9258>`_: [docs] Typo in overloading-guide.rst (`kinow <https://github.com/kinow>`_)
* PR `#9267 <https://github.com/numba/numba/pull/9267>`_: CUDA: Fix dropping of kernels by nvjitlink, by implementing the used list (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9274 <https://github.com/numba/numba/pull/9274>`_: CUDA: Add support for compilation to LTO-IR (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9279 <https://github.com/numba/numba/pull/9279>`_: CUDA: Add support for CUDA 12.0 Windows conda packages (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9292 <https://github.com/numba/numba/pull/9292>`_: CUDA: Switch cooperative groups to use overloads (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9296 <https://github.com/numba/numba/pull/9296>`_: Fix bug when axis is negative and check when axis is invalid (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
* PR `#9301 <https://github.com/numba/numba/pull/9301>`_: Release Notes 0.58.1 for main (`esc <https://github.com/esc>`_)
* PR `#9302 <https://github.com/numba/numba/pull/9302>`_: add missing backtick to example git tag command (`esc <https://github.com/esc>`_)
* PR `#9303 <https://github.com/numba/numba/pull/9303>`_: Add category to warning (`kkokkoros <https://github.com/kkokkoros>`_)
* PR `#9307 <https://github.com/numba/numba/pull/9307>`_: Upgrade to cloudpickle 3.0.0 (`sklam <https://github.com/sklam>`_)
* PR `#9308 <https://github.com/numba/numba/pull/9308>`_: Fix typo in azure ci script (`sklam <https://github.com/sklam>`_)
* PR `#9309 <https://github.com/numba/numba/pull/9309>`_: Continue #9044, prevent compilation on the CPU target when listing tests. (`stuartarchibald <https://github.com/stuartarchibald>`_ `apmasell <https://github.com/apmasell>`_)
* PR `#9310 <https://github.com/numba/numba/pull/9310>`_: Remove Python 3.8 support. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9315 <https://github.com/numba/numba/pull/9315>`_: Fix support for LOAD_FAST_AND_CLEAR (`sklam <https://github.com/sklam>`_)
* PR `#9318 <https://github.com/numba/numba/pull/9318>`_: GPU CI: Test with Python 3.9-3.12 (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9325 <https://github.com/numba/numba/pull/9325>`_: Fix GPUCI (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9326 <https://github.com/numba/numba/pull/9326>`_: Add docs for LOAD_FAST_AND_CLEAR changes  (`sklam <https://github.com/sklam>`_)
* PR `#9330 <https://github.com/numba/numba/pull/9330>`_: Refactor and remove legacy APIs/testing internals. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9331 <https://github.com/numba/numba/pull/9331>`_: Fix Syntax and Deprecation Warnings from 3.12. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9334 <https://github.com/numba/numba/pull/9334>`_: Fix parfor reduction issue with Python 3.12. (`DrTodd13 <https://github.com/DrTodd13>`_)
* PR `#9335 <https://github.com/numba/numba/pull/9335>`_: Add validation capability for user generated towncrier .rst files. (`kc611 <https://github.com/kc611>`_)
* PR `#9336 <https://github.com/numba/numba/pull/9336>`_: Add TargetLibraryInfo pass to CPU LLVM pipeline. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9337 <https://github.com/numba/numba/pull/9337>`_: Revert #8583 which skip tests due to M1 RuntimeDyLd Assertion error (`sklam <https://github.com/sklam>`_)
* PR `#9341 <https://github.com/numba/numba/pull/9341>`_: Add configuration variable to force llvmlite memory manager on / off (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9342 <https://github.com/numba/numba/pull/9342>`_: Fix flake8 checks for v6.1.0 (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9346 <https://github.com/numba/numba/pull/9346>`_: Setting ``NUMBA_CAPTURED_ERRORS=old_style`` will now raise warnings. (`sklam <https://github.com/sklam>`_)
* PR `#9347 <https://github.com/numba/numba/pull/9347>`_: Make Numba's deprecation warnings subclasses of the builtin ones. (`sklam <https://github.com/sklam>`_)
* PR `#9351 <https://github.com/numba/numba/pull/9351>`_: Made Python 3.12 support rst note more verbose (`kc611 <https://github.com/kc611>`_)
* PR `#9352 <https://github.com/numba/numba/pull/9352>`_: Removing object mode fallback from `@jit`. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9353 <https://github.com/numba/numba/pull/9353>`_: Remove `numba.generated_jit` (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9356 <https://github.com/numba/numba/pull/9356>`_: Refactor print tests to avoid NRT leak issue. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9357 <https://github.com/numba/numba/pull/9357>`_: Fix a typo in `_set_init_process_lock` warning. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9358 <https://github.com/numba/numba/pull/9358>`_: Remove note about OpenMP restriction in wheels. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9359 <https://github.com/numba/numba/pull/9359>`_: Fix test_jit_module test against objmode fallback. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9360 <https://github.com/numba/numba/pull/9360>`_: AzureCI changes. RVSDG test config should still test its assigned test slice (`sklam <https://github.com/sklam>`_)
* PR `#9362 <https://github.com/numba/numba/pull/9362>`_: Fix ``np.MachAr`` warning matching in test. (`sklam <https://github.com/sklam>`_)
* PR `#9364 <https://github.com/numba/numba/pull/9364>`_: Release notes fixes for appropriate Towncrier header underlines (`kc611 <https://github.com/kc611>`_)
* PR `#9367 <https://github.com/numba/numba/pull/9367>`_: Document release notes generation (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9368 <https://github.com/numba/numba/pull/9368>`_: Added 0.59.0 release notes (`kc611 <https://github.com/kc611>`_)
* PR `#9369 <https://github.com/numba/numba/pull/9369>`_: Fix release notes link in bug report template (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9378 <https://github.com/numba/numba/pull/9378>`_: Add `float(<string literal>)` ctor (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
* PR `#9394 <https://github.com/numba/numba/pull/9394>`_: fix `TypeError: '_DeviceContextManager' object is not callable` (`i7878 <https://github.com/i7878>`_)
* PR `#9402 <https://github.com/numba/numba/pull/9402>`_: Doc updates for 0.59 final (`sklam <https://github.com/sklam>`_)
* PR `#9403 <https://github.com/numba/numba/pull/9403>`_: Fix test isolation for stateful configurations in the testsuite (`sklam <https://github.com/sklam>`_)
* PR `#9404 <https://github.com/numba/numba/pull/9404>`_: Fix skipped test stderr change for Python 3.12.1. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9407 <https://github.com/numba/numba/pull/9407>`_: Fix incorrect cycle detection (`sklam <https://github.com/sklam>`_)
* PR `#9411 <https://github.com/numba/numba/pull/9411>`_: Doc updates for 0.59.0 final. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9416 <https://github.com/numba/numba/pull/9416>`_: Add math.log2 support (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
* PR `#9417 <https://github.com/numba/numba/pull/9417>`_: Add np.log* bindings for CUDA (`gmarkall <https://github.com/gmarkall>`_ `guilhermeleobas <https://github.com/guilhermeleobas>`_)
* PR `#9425 <https://github.com/numba/numba/pull/9425>`_: Post release for 0.59.0 (`sklam <https://github.com/sklam>`_)
* PR `#9436 <https://github.com/numba/numba/pull/9436>`_: Add timing and junit xml output to testsuite (`sklam <https://github.com/sklam>`_)
* PR `#9437 <https://github.com/numba/numba/pull/9437>`_: Remove dependencies between Numba's Cpython and NumPy module (`kc611 <https://github.com/kc611>`_)
* PR `#9438 <https://github.com/numba/numba/pull/9438>`_: Add math.nextafter support for nopython mode. (`groutr <https://github.com/groutr>`_)
* PR `#9445 <https://github.com/numba/numba/pull/9445>`_: Fix #9427 np.searchsorted on datetime64 (`sklam <https://github.com/sklam>`_)
* PR `#9447 <https://github.com/numba/numba/pull/9447>`_: Fix Issue #9432, caching of kernels using target-specific overloads (such as CG sync) (`gmarkall <https://github.com/gmarkall>`_ `sklam <https://github.com/sklam>`_)
* PR `#9448 <https://github.com/numba/numba/pull/9448>`_: Fix `np.searchsorted` regression (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
* PR `#9449 <https://github.com/numba/numba/pull/9449>`_: Remove deprecated CondaEnvironment@1 (`sklam <https://github.com/sklam>`_)
* PR `#9450 <https://github.com/numba/numba/pull/9450>`_: Fix gpuci versions (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9454 <https://github.com/numba/numba/pull/9454>`_: Don't attempt to register overloads that aren't for this target in `BaseContext` and related fixes (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9455 <https://github.com/numba/numba/pull/9455>`_: Support datetime types in `isinstance()` (`sklam <https://github.com/sklam>`_)
* PR `#9456 <https://github.com/numba/numba/pull/9456>`_: Update release checklist (`sklam <https://github.com/sklam>`_)
* PR `#9459 <https://github.com/numba/numba/pull/9459>`_: Support pep695 type param syntax (`sklam <https://github.com/sklam>`_)
* PR `#9466 <https://github.com/numba/numba/pull/9466>`_: Numpy 2.0 binary support testing (`kc611 <https://github.com/kc611>`_)
* PR `#9468 <https://github.com/numba/numba/pull/9468>`_: adding git-copy.py script (`esc <https://github.com/esc>`_)
* PR `#9482 <https://github.com/numba/numba/pull/9482>`_: Add support for `sys.monitoring` events. (`stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#9487 <https://github.com/numba/numba/pull/9487>`_: Add CUDA 12 conda installation docs (`gmarkall <https://github.com/gmarkall>`_ `bdice <https://github.com/bdice>`_)
* PR `#9488 <https://github.com/numba/numba/pull/9488>`_: Update overview.rst (`jftsang <https://github.com/jftsang>`_)
* PR `#9491 <https://github.com/numba/numba/pull/9491>`_: Fix non-deterministic bug caused by unstableness in SSA  (`sklam <https://github.com/sklam>`_ `loicdtx <https://github.com/loicdtx>`_)
* PR `#9502 <https://github.com/numba/numba/pull/9502>`_: Post release task for 0.59.1 (`sklam <https://github.com/sklam>`_)
* PR `#9504 <https://github.com/numba/numba/pull/9504>`_: added np.size() overload and added tests (`shourya5 <https://github.com/shourya5>`_)
* PR `#9521 <https://github.com/numba/numba/pull/9521>`_: Support binop reduction. (`DrTodd13 <https://github.com/DrTodd13>`_)
* PR `#9531 <https://github.com/numba/numba/pull/9531>`_: Module pass manager: Don't add passes for unsupported LLVM versions (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9532 <https://github.com/numba/numba/pull/9532>`_: Fix hoisting bug to exclude variables used in containers or calls. (`DrTodd13 <https://github.com/DrTodd13>`_)
* PR `#9539 <https://github.com/numba/numba/pull/9539>`_: Revert PR #6870 `numba.core.retarget` (`sklam <https://github.com/sklam>`_)
* PR `#9549 <https://github.com/numba/numba/pull/9549>`_: Make new_style the default error capturing mode (`sklam <https://github.com/sklam>`_ `gmarkall <https://github.com/gmarkall>`_)

Authors:

* `apmasell <https://github.com/apmasell>`_
* `bdice <https://github.com/bdice>`_
* `dlee992 <https://github.com/dlee992>`_
* `DrTodd13 <https://github.com/DrTodd13>`_
* `esc <https://github.com/esc>`_
* `gmarkall <https://github.com/gmarkall>`_
* `groutr <https://github.com/groutr>`_
* `guilhermeleobas <https://github.com/guilhermeleobas>`_
* `hellozee <https://github.com/hellozee>`_
* `i7878 <https://github.com/i7878>`_
* `jftsang <https://github.com/jftsang>`_
* `kc611 <https://github.com/kc611>`_
* `kinow <https://github.com/kinow>`_
* `kkokkoros <https://github.com/kkokkoros>`_
* `KrisMinchev <https://github.com/KrisMinchev>`_
* `loicdtx <https://github.com/loicdtx>`_
* `munahaf <https://github.com/munahaf>`_
* `njriasan <https://github.com/njriasan>`_
* `rjenc29 <https://github.com/rjenc29>`_
* `saulshanabrook <https://github.com/saulshanabrook>`_
* `shourya5 <https://github.com/shourya5>`_
* `sklam <https://github.com/sklam>`_
* `SridharCR <https://github.com/SridharCR>`_
* `stuartarchibald <https://github.com/stuartarchibald>`_
* `Tialo <https://github.com/Tialo>`_
```
Am pruning it now.

**Comment by esc** [comment]
I just tried with the `dev` tag and there were two additional PRs. The first one, not sure why it wasn't picked up, the second, is because this PR was merged AFTER the initial generation of the list.

* PR `#9367 <https://github.com/numba/numba/pull/9367>`_: Document release notes generation (`gmarkall <https://github.com/gmarkall>`_)
* PR `#9559 <https://github.com/numba/numba/pull/9559>`_: Update version support table 0.60 (`esc <https://github.com/esc>`_)

**Comment by esc** [comment]
> Putting in the dev tag `0.59.0dev0` as the beginning give the following list of PRs:

I think that will give you all commits from the 0.59 AND 0.60 release. I think you will need the `0.60.0dev0` tag.

**Comment by kc611** [comment]
Okay so the pruned PR list and the list in this PR is almost the same. As you pointed out #9367 wasn't in the pruned PR list not sure why, but it is already within this list. I'll add #9559 within this list shortly, the only other PRs that were in the pruned list were #9368 which I'm not sure whould be a part of this changelog?

**Comment by esc** [comment]
Yeah, it seems like both #9367 and #9368 were merged BEFORE the tag `0.60.0dev0` was set:

Screenshot 2024-05-08 at 09 55 50

**Comment by esc** [comment]
@kc611 looks good thank you, just waiting on CI now.

**Pull request merged**